service/controller/resource/tcnp/template/template_main_auto_scaling_group.go

```diff
@@ -16,7 +16,14 @@ const TemplateMainAutoScalingGroup = `
       DesiredCapacity: {{ .AutoScalingGroup.DesiredCapacity }}
       MinSize: {{ .AutoScalingGroup.MinSize }}
       MaxSize: {{ .AutoScalingGroup.MaxSize }}
-      LaunchConfigurationName: !Ref NodePoolLaunchConfiguration
+      MixedInstancesPolicy: 
+        LaunchTemplate:
+          LaunchTemplateSpecification:
+            LaunchTemplateName: !Ref NodePoolLaunchTemplate
+            Version:
+              Fn::GetAtt
+              - NodePoolLaunchTemplate
+              - LatestVersionNumber
```

Comment:
Why is this `LatestVersionNumber` needed and what does it do?

Reply:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-autoscalinggroup-launchtemplatespecification.html